components/places/src/api/places_api.rs

```diff
@@ -267,7 +271,8 @@ impl PlacesApi {
         Ok(sync_ping)
     }
 
-    // TODO: reduce duplication with above
+    // NOTE: This is to be deprecated
+    // #[deprecated(note = "Please use the .sync() API to sync both stores")]
```

Comment:
@grigoryk is the expert here, but I think deprecating this would create some API churn for Android Components...it has separate `PlacesHistoryStorage` and `PlacesBookmarksStorage` classes that call `syncHistory` and `syncBookmarks` separately now, though backed by the same `PlacesConnection`.

But maybe the sync manager design would mean removing `sync()` from those classes entirely, and having its high-level sync manager call ours.

Reply:
Yeah - I softened the wording in that comment and I don't see a good reason to include all the ffi changes before landing this, so we can have that discussion with grisha in a followup.